service/dap/server.go

```diff
@@ -2968,6 +3057,73 @@ func (s *Server) doRunCommand(command string, asyncSetupDone chan struct{}) {
 	}
 }
 
+func (s *Server) getHitBreakpointNameAndId(stoppedGoroutineID int, state *api.DebuggerState) (string, int) {
+	if g, err := s.debugger.FindGoroutine(stoppedGoroutineID); err == nil && g != nil && g.Thread != nil && g.Thread.Breakpoint() != nil {
+		if bp := g.Thread.Breakpoint().Breakpoint; bp != nil {
+			return bp.Name, bp.LogicalID
+		}
+	}
+	if state.CurrentThread != nil && state.CurrentThread.Breakpoint != nil {
+		return state.CurrentThread.Breakpoint.Name, state.CurrentThread.Breakpoint.ID
+	}
+	return "", -1
+}
+
+func (s *Server) run(command string, asyncSetupDone chan struct{}) (*api.DebuggerState, error) {
+	s.setRunning(true)
+	defer s.setRunning(false)
+
+	// Clear any manual stop requests that came in before we started running.
+	s.debugger.CheckAndClearManualStopRequest()
+	var state *api.DebuggerState
+	var err error
+	for {
+		state, err = s.resume(command, asyncSetupDone)
+		if state == nil || err != nil {
```

Comment:
I think you would catch processExited state either as error or as stopReason below, but it might be cleaner to be explicit about it and just break if processExited() is true.

Reply:
> I think you would catch processExited state either as error or as stopReason below, but it might be cleaner to be explicit about it and just break if processExited() is true.

Btw,  I think at some point stopReason was not always Exited when it had to be (maybe because it was added latere), which is why processExited looks at state.Exited.